src/EntityFramework/Core/EntityKey.cs

```diff
@@ -1219,6 +1219,7 @@ internal void ValidateEntityKey(MetadataWorkspace workspace, EntitySet entitySet
         // based on whether the key is a singleton, composite, or temporary.
         // </summary>
         // <param name="isTemporary"> whether we expect this EntityKey to be marked temporary </param>
+        [SuppressMessage("Microsoft.Performance", "CA1800:DoNotCastUnnecessarily", Justification = "Only used in debug mode.")]
```

Comment:
Nit: assigning (EntitySet)entitySet to a variable would have been easier :smile: